src/WebJobs.Script.WebHost/Program.cs

```diff
@@ -84,19 +84,35 @@ private static void InitializeProcess()
             {
                 // Linux containers always start out in placeholder mode
                 SystemEnvironment.Instance.SetEnvironmentVariable(EnvironmentSettingNames.AzureWebsitePlaceholderMode, "1");
+                AppDomain.CurrentDomain.UnhandledException += CurrentDomainOnUnhandledException;
             }
 
             // Some environments only set the auth key. Ensure that is used as the encryption key if that is not set
             string authEncryptionKey = SystemEnvironment.Instance.GetEnvironmentVariable(EnvironmentSettingNames.WebSiteAuthEncryptionKey);
             if (authEncryptionKey != null &&
-                SystemEnvironment.Instance.GetEnvironmentVariable(DataProtectionCostants.AzureWebsiteEnvironmentMachineKey) == null)
+                SystemEnvironment.Instance.GetEnvironmentVariable(DataProtectionConstants.AzureWebsiteEnvironmentMachineKey) == null)
             {
-                SystemEnvironment.Instance.SetEnvironmentVariable(DataProtectionCostants.AzureWebsiteEnvironmentMachineKey, authEncryptionKey);
+                SystemEnvironment.Instance.SetEnvironmentVariable(DataProtectionConstants.AzureWebsiteEnvironmentMachineKey, authEncryptionKey);
             }
 
             ConfigureMinimumThreads(SystemEnvironment.Instance.IsWindowsConsumption());
         }
 
+        private static void CurrentDomainOnUnhandledException(object sender, UnhandledExceptionEventArgs e)
+        {
+            // Fallback console logs in case kusto logging fails.
+            Console.WriteLine($"{nameof(CurrentDomainOnUnhandledException)}: {e.ExceptionObject}");
+
+            var linuxContainerEventGenerator = new LinuxContainerEventGenerator(SystemEnvironment.Instance);
```

Comment:
Consider adding a static method LinuxContainerEventGenerator.LogUnhandledException(Exception ex) that does this and the above, so this method is just a one liner?